File: Day 1 - Arrays I/53. Maximum Subarray/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    long long maxSubArray(vector<int>& nums) {
        long long maxSum = INT_MIN;
        long long sum = 0;
        for (int i = 0; i < nums.size(); i++) {
            cout << sum << " ";
            sum += nums[i];
            maxSum = max(maxSum, sum);
            if (sum < 0) sum = 0;
        }
        return maxSum;
    }
};

int main() {
    Solution solution;
    vector<int> nums = {-7, -8, -16, -4 ,-8, -5, -7, -11, -10, -12, -4, -6, -4, -16, -10 };
    cout << solution.maxSubArray(nums) << endl;
    return 0;
}

File: Day 1 - Arrays I/53. Maximum Subarray/Solution.py
class Solution:
    def maxSubArray(self, nums: [int]) -> int:
        sumN = 0
        maxS = -9999999

        for i in nums:
            sumN += i

            if (sumN > maxS):
                maxS = sumN

            if sumN < 0:
                sumN = 0

        return maxS

if __name__ == '__main__':
    s = Solution()
    print(s.maxSubArray([-2,1,-3,4,-1,2,1,-5,4]))
    print(s.maxSubArray([1]))
    print(s.maxSubArray([5,4,-1,7,8]))